File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/Signal.java
/*
 * Copyright (c) 2018 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models;

import android.support.annotation.NonNull;

import com.fishwaffle.natureremo.controller.NatureRemo;
import com.fishwaffle.natureremo.controller.SignalImage;

import java.io.Serializable;

import static com.fishwaffle.natureremo.controller.Util.requireNonNull;

public class Signal implements Serializable {
    public String id;
    public String name;
    public String image;

    public Signal Update(String token, SignalImage image, String name) {
        return NatureRemo.Signals_Signal_Post(token, id, requireNonNull(image, SignalImage.valueOf(this.image)), requireNonNull(name, this.name));
    }

    public void Send(String token) {
        NatureRemo.Signals_Signal_Send_Post(token, id);
    }

    public void Delete(String token) {
        NatureRemo.Signals_Signal_Delete_Post(token, id);
    }

    @NonNull
    @Override
    public String toString() {
        return "Signal{" + "id='" + id + '\'' + ", name='" + name + '\'' + ", image='" + image + '\'' + '}';
    }
}

File: natureremocontroller/src/main/java/com/fishwaffle/natureremo/controller/models/Signal.kt
/*
 * Copyright (c) 2019 FishWaffle.
 */

package com.fishwaffle.natureremo.controller.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fishwaffle.natureremo.controller.NatureRemo
import com.fishwaffle.natureremo.controller.NatureRemo.isIgnoreUnknown
import com.fishwaffle.natureremo.controller.SignalImage
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = isIgnoreUnknown)
data class Signal(var id: String?, var name: String?, var image: String?) : Serializable {

    fun update(token: String, image: SignalImage, name: String): Signal? {
        return NatureRemo.signalsSignalPost(token, id!!, image, name)
    }

    fun send(token: String) {
        NatureRemo.signalsSignalSendPost(token, id!!)
    }

    fun delete(token: String) {
        NatureRemo.signalsSignalDeletePost(token, id!!)
    }


}
